src/patterns/ProgressWizard/components/InnerProgressWizard.tsx

```diff
@@ -0,0 +1,183 @@
+    customHeader?: ReactNode;
+    hideNavigation?: boolean;
+    hideProgressIndicator?: boolean;
+    buttonLabels: ButtonLabels;
+    action?: string;
+    isFirstStep: boolean;
+    isLastStep: boolean;
+    className?: string;
+    formId: string;
+  };
+
+/**
+ * Internal layout component for ProgressWizard. Renders the wizard UI, progress indicator, step content, and footer.
+ *
+ * This component is used internally by ProgressWizard and should not be used directly by consumers.
+ *
+ * @param props - InnerProgressWizardProps
+ * @returns ReactElement with wizard layout
+ */
+const InnerProgressWizard = forwardRef<HTMLDivElement, InnerProgressWizardProps>((props, ref) => {
+  const {
+    steps,
+    currentStepIndex,
+    setCurrentStepIndex,
+    customHeader,
+    hideNavigation,
+    hideProgressIndicator,
+    buttonLabels,
+    loadingState,
+    setLoadingState,
+    action,
+    isFirstStep,
+    isLastStep,
+
+    handleContinue,
+    handleBack,
+    handleSubmit,
+    handleCancel,
+
+    className,
+    formId,
+
+    ...rest
+  } = props;
+
+  const { className: baseClassName, ...commonProps } = getCommonProps(rest, 'ProgressWizard');
+  const currentStep = useMemo(
+    () =>
+      steps[currentStepIndex] ?? {
+        id: '0',
+        label: '',
+        componentFactory: () => <span>No step found at index {currentStepIndex}</span>,
+      },
+    [steps, currentStepIndex],
+  );
+  const formMethods = useFormContext();
+
+  useEffect(() => {
+    if (currentStep && currentStep.hiddenFields) {
+      handleHiddenFields(Object.keys(formMethods.getValues()), currentStep.hiddenFields);
+    }
+  }, [currentStep, formMethods]);
+
+  const { registerProgressWizardInput } = useProgressWizardForm();
+  const currentStepComponent = useMemo(
+    () =>
+      cloneElement(
+        currentStep?.componentFactory({
+          ...formMethods,
+          registerProgressWizardInput: (fieldName, options) =>
+            registerProgressWizardInput(fieldName, options, currentStep.id),
+          currentStepIndex,
+          setCurrentStepIndex,
+          loadingState,
+          setLoadingState,
+          handlers: {
+            handleContinue,
+            handleBack,
+            handleSubmit,
+            handleCancel,
+          },
+        }),
+        {
+          key: currentStep.id,
+        },
+      ),
+    [
+      currentStep,
+      currentStepIndex,
+      formMethods,
+      loadingState,
+      registerProgressWizardInput,
+      setCurrentStepIndex,
+      setLoadingState,
+      handleContinue,
+      handleBack,
+      handleSubmit,
+      handleCancel,
+    ],
+  );
+
+  return (
+    <section {...commonProps} className={classNames(baseClassName, className)} ref={ref} aria-label="Form Wizard">
+      <div className={`${baseClassName}__logo`}>
+        <Icon icon="PhillipsLogo" height={32} width={120} />
+      </div>
+      {customHeader}
+      <nav aria-label="Progress">
+        {!hideProgressIndicator ? (
```

Comment:
If we are not rendering anything else in the nav element, perhaps we raise this ternary up to include the whole nav element or null?